src/plugins/core/cell.ts

```diff
@@ -58,6 +77,484 @@ export class CellPlugin extends CorePlugin<CoreState> implements CoreState {
+    options?: ClipboardOptions | undefined
+  ) {
+    if (!content.isCutOperation) {
+      this.pasteFromCopy(target, content, getters, dispatch, options);
+    } else {
+      this.pasteFromCut(target, content, getters, dispatch, options);
+    }
+    const height = content.cells.length;
+    const width = content.cells[0].length;
+    if (options?.selectTarget) {
+      this.selectPastedZone(width, height, content.isCutOperation, target);
+    }
+  }
+
+  pasteFromCopy(
+    target: Zone[],
+    content: ClipboardContent,
+    getters: Getters,
+    dispatch: CommandDispatcher["dispatch"],
+    options?: ClipboardOptions
+  ) {
+    const sheetId = getters.getActiveSheetId();
+    if (target.length === 1) {
+      // in this specific case, due to the isPasteAllowed function:
+      // state.cells can contains several cells.
+      // So if the target zone is larger than the copied zone,
+      // we duplicate each cells as many times as possible to fill the zone.
+      const height = content.cells.length;
+      const width = content.cells[0].length;
+      const pasteZones = pastedZones(target, width, height);
+      for (const zone of pasteZones) {
+        this.pasteZone(zone.left, zone.top, content, getters, dispatch, options);
+      }
+    } else {
+      // in this case, due to the isPasteAllowed function: state.cells contains
+      // only one cell
+      for (const zone of target) {
+        for (let col = zone.left; col <= zone.right; col++) {
+          for (let row = zone.top; row <= zone.bottom; row++) {
+            this.pasteZone(col, row, content, getters, dispatch, options);
+          }
+        }
+      }
+    }
+    if (options?.pasteOption === undefined) {
+      this.pasteCopiedTables(sheetId, target, content, dispatch);
+    }
+  }
+
+  pasteFromCut(
+    target: Zone[],
+    content: ClipboardContent,
+    getters: Getters,
+    dispatch: CommandDispatcher["dispatch"],
+    options?: ClipboardOptions
+  ) {
+    this.clearClippedZones(content, dispatch);
+    const selection = target[0];
+    const sheetId = getters.getActiveSheetId();
+    this.pasteZone(selection.left, selection.top, content, getters, dispatch, options);
+    dispatch("MOVE_RANGES", {
+      target: content.zones,
+      sheetId: content.sheetId,
+      targetSheetId: getters.getActiveSheetId(),
+      col: selection.left,
+      row: selection.top,
+    });
+
+    for (const filterTable of content.tables) {
+      dispatch("REMOVE_FILTER_TABLE", {
+        sheetId: getters.getActiveSheetId(),
+        target: [filterTable.zone],
+      });
+    }
+    this.pasteCopiedTables(sheetId, target, content, dispatch);
+    content.cells.forEach((row) => {
+      row.forEach((c) => {
+        if (c.cell) {
+          c.cell = undefined;
+        }
+      });
+    });
+  }
+
+  /**
+   * Add columns and/or rows to ensure that col + width and row + height are still
+   * in the sheet
+   */
+  protected addMissingDimensions(
```

Comment:
each clipboard handler could return its own target zone.
The clipboard can then add the missing cols/rows and select the union of zone at the end